app/models/classification.rb

```diff
@@ -46,6 +46,27 @@ class Classification < ApplicationRecord
 
   FIXTURE_FILE = FIXTURE_DIR.join("classifications.yml")
 
+  def self.parent_classifications
+    where(:classifications => {:parent_id => 0})
+  end
+
+  def self.hash_all_by_type_and_name_with_mapped_labels(conditions = {})
+    ret = {}
+
+    condition_for_mapped_tags = ContainerLabelTagMapping::TAG_PREFIXES.map { "tags.name LIKE ?" }.join(' OR ')
+
+    tag_values = ContainerLabelTagMapping::TAG_PREFIXES.map { |x| "#{x}%:%" }
+    where(conditions).parent_classifications.includes(:tag).references(:tag).where(condition_for_mapped_tags, *tag_values).each do |c|
```

Comment:
Could this use `joins(:tag).merge(Tag.controlled_by_mapping)` instead of directly querying on TAG_PREFIXES?
I'd like to not leak the fact it's currently implemented as prefix-based query